src/components/Books.js

```diff
@@ -29,20 +29,48 @@ const useGetBook = (bookId) => {
     };
 };
 
-const updateBookStatus = async (bookId, checkedOutBy) => {
+const updateBookToReturned = async (bookId, userProfile) => {
+    const event = {
+        timestamp: new Date(Date.now()).toJSON(),
+        name: userProfile.name,
+        email: userProfile.email
+    };
     return await documentClient.update({
         TableName: process.env.REACT_APP_BOOK_TABLE,
         Key: {
             bookId
         },
-        UpdateExpression: "set checkedOutBy=:l",
+        UpdateExpression: "set checkedOutBy=:l, returnEvents=list_append(if_not_exists(returnEvents, :emptyList), :returnEvents)",
         ExpressionAttributeValues: {
-            ':l': checkedOutBy
+            ':l': null,
+            ':emptyList': [],
+            ':returnEvents': [event]
         },
         ReturnValues: 'UPDATED_NEW'
     }).promise();
 };
 
+const updateBookToCheckedOut = async (bookId, userProfile) => {
+    const event = {
+        timestamp: new Date(Date.now()).toJSON(),
+        name: userProfile.name,
+        email: userProfile.email
+    };
+    return await documentClient.update({
```

Comment:
Having the only `await` in an `async` function be part of the return statement feels like a code smell and has definitely been a linter error for me in the past. The equivalent would be dropping both of these keywords and still `await-ing` the returned value. https://eslint.org/docs/rules/no-return-await

Reply:
I've seen that linter error in the past as well. For a while there was a concern that awaiting a return was inefficient. I think we can go either way, but one thing about the explicit async/await is that it makes it clear that this function returns a Promise and needs to be awaited by the caller.